app/views.py

```diff
@@ -244,6 +246,34 @@ def recent_users():
                               'results': users,
                               'query': ''})
 
+@views.route('/activity', methods=['GET', 'POST'])
+@login_required
+def activity():
+    '''
+    View for the activity feed of recent events.
+    '''
+
+    events = Event.query_in_deployment().order_by(desc(Event.created_at)).limit(10).all()
+    shared_message_form = SharedMessageForm()
+
+    if request.method == 'POST':
+        if not current_user.display_in_search:
+            flash(gettext(u'We need your name before you can post a message.'), 'error')
```

Comment:
An alternative here is to add the message to the database but simply not display it if the user's name is blank... Or we could substitute "Mysterious Person" as their name in the activity feed! I suspect the number of mysterious people will plummet once we make first/last name part of the signup process anyways?